src/main/java/org/jitsi/jicofo/recording/jibri/JibriRecorder.java

```diff
@@ -189,20 +194,32 @@ public boolean accept(Packet packet)
     /**
      * Sends an IQ to the given Jibri instance and asks it to start recording.
      */
-    private XMPPError startJibri(final String    jibriJid,
-                                 final String    roomName,
-                                 final String    streamId)
-    throws SmackException.NoResponseException
-    {
+    synchronized private XMPPError startJibri(
+        final String jibriJid
+    ) throws SmackException.NoResponseException {
+        return startJibri(jibriJid, "");
+    }
+
+    /**
+     * Sends an IQ to the given Jibri instance and asks it to start recording.
+     */
+    synchronized private XMPPError startJibri(
+        final String jibriJid,
+        final String streamId
+    ) throws SmackException.NoResponseException {
         JibriIq startIq = new JibriIq();
         startIq.setTo(jibriJid);
         startIq.setType(IQ.Type.SET);
         startIq.setAction(JibriIq.Action.START);
 
-        startIq.setStreamId(streamId);
+        if ("".equals(streamId))
```

Comment:
The if works: if streamId equals to empty string then set it in the iq, but otherwise it is left empty.